flake.nix

```diff
@@ -0,0 +1,108 @@
+{
+  description = "slugger: Clean URI slugs for Haskell";
+
+  nixConfig.bash-prompt = "[nix]λ ";
+
+  inputs = {
+    nixpkgs.url = "nixpkgs/nixos-21.05";
+
+    flake-utils = {
+      url = "github:numtide/flake-utils";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+  };
+
+  outputs = { flake-utils, nixpkgs, self }:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        config = {};
+
+        compilerVersion = "884";
+        compiler = "ghc" + compilerVersion;
+        overlays = [
+          (final: prev: {
+            haskell-language-server = prev.haskell-language-server.override {
+              supportedGhcVersions = [ compilerVersion ];
+            };
+
+            myHaskellPackages = prev.haskell.packages.${compiler}.override {
+              overrides = hpFinal: hpPrev: {
+                slugger = hpPrev.callCabal2nix "slugger" ./. {};
+
+                hspec = hpPrev.callHackage "hspec" "2.8.1" {}; # see slugger.cabal build-depends
+                hspec-core = hpPrev.callHackage "hspec-core" "2.8.1" {}; # because hspec
+                hspec-discover = hpPrev.callHackage "hspec-discover" "2.8.1" {}; # because hspec
```

Comment:
I guess these are because of the tight version bounds I mentioned on another comment; all are very common packages so I'd be surprised if they don't work on most supported GHC's straightaway.

Reply:
Also, you don't seem to be using `hspec-discover` either.

Reply:
> Also, you don't seem to be using `hspec-discover` either.

Since `hspec` `2.8.1` does not exist in that pinned version of nixpkgs and I'm caling `callHackage`, `hspec-discover` `2.8.1` – a dependency of `hspec` – also doesn't exist, and so an error is thrown. Both `hspec-core` and `hspec-discover` are included here because `hspec` won't build without them (the deps are `==`)

Reply:
> I guess these are because of the tight version bounds I mentioned on another comment; all are very common packages so I'd be surprised if they don't work on most supported GHC's straightaway.

I'll play around and see if loosening the version bounds will clean this up. I bet you're right